spec/unit/sensu_client_config_spec.rb

```diff
@@ -62,4 +62,22 @@
       end
     end
   end
+  describe 'http_socket' do
```

Comment:
Could you add a test that reads a fixture file and sees that the json created matches like what is done in `spec/defines/sensu_check_spec.rb`.

Reply:
@ghoneycutt sorry, I spent 2 hours in trying to make it and I didn't succeed, so I prefer to raise hand before engaging further efforts.
First, I'm not 100% sure on where you want the test, on the sensu_client_config type (here) on n the sensu::client class.
I've tried with both , but haven't any usable resource to compare directly  to the final generated json file, in spec/classes/sensu_client_spec.rb (where I could use an approach similar to the spec/defines/sensu_check_spec.rb) I can only tests parameters passed to the the sensu_client_config type, and is is already done.
In spec/unit/sensu_client_config_spec.rb I don't know what to compare to the content of the fixture file, I can test single params, but haven't found a way to do something based on the whole file.
IMHO current tests, and tested implementation on real VM, could be enough, considering that also for other params of sensu_client_config do don't tests agains fixture json files.
But I leave to you the final word here and, eventually, the support to finalise the tests.